script/wait_for_file.sh

```diff
@@ -0,0 +1,8 @@
+wait_for_file() {
+  local file="$1"; shift
+  local wait_seconds="${1:-10}"; shift # 10 seconds as default timeout
+
+  until test $((wait_seconds--)) -eq 0 -o -f "$file" ; do sleep 1; done
+
+  ((++wait_seconds))
```

Comment:
And, obviously, I didn't know what the return value of `(( expression ))` is (From manual: If the value of the  expression  is non-zero, the return status is 0; otherwise the return status is 1.). So your code was correct, but then it even more seems magical. I'd prefer the more explicit test in the end.